### ISSUE TITLE
ActivityWindow updates will cause CoreAnimation errors

### ISSUE THREAD

**Issue by Eitot** [ViennaRSS/vienna-rss#833]
When the activity window is open and the content refreshed (command–R), then scrolling in the table view of the activity window produces many error messages of this kind:
```
2017-03-20 21:25:24.735189 Vienna[36966:11089287] Uncommitted CATransaction. Set CA_DEBUG_TRANSACTIONS=1 in environment to debug.
```

Adding the environment variable as suggested, will produce this stack trace (on branch `master`):
```
2017-03-20 21:46:46.234442 Vienna[38270:11219605] Unexpected outstanding background CATransaction
CoreAnimation: warning, encountered thread with uncommitted CATransaction; created by:
0   QuartzCore                          0x00007fff971b5e5b _ZN2CA11Transaction4pushEv + 331
1   QuartzCore                          0x00007fff971b621f _ZN2CA11Transaction15ensure_implicitEv + 281
2   QuartzCore                          0x00007fff972c688f _ZN2CA5Layer13thread_flags_EPNS_11TransactionE + 37
3   QuartzCore                          0x00007fff972c8812 _ZN2CA5Layer4markEPNS_11TransactionEjj + 68
4   QuartzCore                          0x00007fff972d187f _ZN2CA5Layer25set_needs_display_in_rectERK6CGRect + 311
5   QuartzCore                          0x00007fff972d1950 -[CALayer setNeedsDisplayInRect:] + 25
6   AppKit                              0x00007fff8efa43e9 _NSBackingLayerSetNeedsDisplayInRect + 319
7   AppKit                              0x00007fff8efa42a4 -[_NSBackingLayer setNeedsDisplayInRect:] + 61
8   QuartzCore                          0x00007fff972d1931 -[CALayer setNeedsDisplay] + 62
9   AppKit                              0x00007fff8efa4e1a -[NSView(NSInternal) _setLayerNeedsDisplayInViewRect:] + 473
10  AppKit                              0x00007fff8ef74cfd NSViewSetNeedsDisplayInRect + 897
11  AppKit                              0x00007fff8ef74973 -[NSView setNeedsDisplayInRect:] + 48
12  WebKitLegacy                        0x00007fff9974a884 -[NSView(WebSetNeedsDisplayInRect) _web_setNeedsDisplayInRect:] + 468
13  AppKit                              0x00007fff8f2f2c79 -[_NSMirrorDocumentView documentViewNeedsDisplayInRect:] + 268
14  CoreFoundation                      0x00007fff9147c9fc __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
15  CoreFoundation                      0x00007fff9147c8fb _CFXRegistrationPost + 427
16  CoreFoundation                      0x00007fff9147c662 ___CFXNotificationPost_block_invoke + 50
17  CoreFoundation                      0x00007fff9143ad12 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 2018
18  CoreFoundation                      0x00007fff91439cfb _CFXNotificationPost + 667
19  Foundation                          0x00007fff92e7da37 -[NSNotificationCenter postNotificationName:object:userInfo:] + 66
20  AppKit                              0x00007fff8ef74a92 NSViewSetNeedsDisplayInRect + 278
21  AppKit                              0x00007fff8ef74973 -[NSView setNeedsDisplayInRect:] + 48
22  WebKitLegacy                        0x00007fff9974a884 -[NSView(WebSetNeedsDisplayInRect) _web_setNeedsDisplayInRect:] + 468
23  AppKit                              0x00007fff8f09badd -[NSTableView _tileAndRedisplayAll] + 231
24  Vienna                              0x0000000100035256 -[ExtendedTableView reloadData] + 54
25  Vienna                              0x000000010005c337 -[ActivityViewer reloadTable] + 343
26  Vienna                              0x000000010005c48a -[ActivityViewer handleLogChange:] + 58
27  CoreFoundation                      0x00007fff9147c9fc __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
28  CoreFoundation                      0x00007fff9147c8fb _CFXRegistrationPost + 427
29  CoreFoundation                      0x00007fff9147c662 ___CFXNotificationPost_block_invoke + 50
30  CoreFoundation                      0x00007fff9143ad12 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 2018
31  CoreFoundation                      0x00007fff91439cfb _CFXNotificationPost + 667
32  Foundation                          0x00007fff92e7da37 -[NSNotificationCenter postNotificationName:object:userInfo:] + 66
33  Vienna                              0x000000010005ac46 -[ActivityItem setStatus:] + 150
34  Vienna                              0x00000001000ca635 __41-[RefreshManager folderRefreshCompleted:]_block_invoke + 1525
35  libdispatch.dylib                   0x00000001004876e5 _dispatch_call_block_and_release + 12
36  libdispatch.dylib                   0x000000010047df5c _dispatch_client_callout + 8
37  libdispatch.dylib                   0x0000000100495a59 _dispatch_queue_serial_drain + 205
38  libdispatch.dylib                   0x0000000100487489 _dispatch_queue_invoke + 1174
39  libdispatch.dylib                   0x000000010048e707 _dispatch_queue_override_invoke + 656
40  libdispatch.dylib                   0x00000001004801d7 _dispatch_root_queue_drain + 671
41  libdispatch.dylib                   0x000000010047fee8 _dispatch_worker_thread3 + 114
42  libsystem_pthread.dylib             0x00000001004f589a _pthread_wqthread + 1299
43  libsystem_pthread.dylib             0x00000001004f5375 start_wqthread + 13
(lldb) 
```

A web search suggests that this error is caused by a race condition involving a CoreAnimation queue (which is executed on the main thread). Looking at the trace, I think the method `-(void)folderRefreshCompleted:(ASIHTTPRequest *)connector` is partially responsible. It has a dispatch queue, so it might conflict with the CoreAnimation queue on the main thread.

**Comment by josh64x2** [comment]
Great investigation! I will try to take a look at it.

**Comment by Eitot** [comment]
I have just tested this:
```objc
dispatch_async(dispatch_get_main_queue(), ^{
    [connectorItem appendDetail:NSLocalizedString(@"Got HTTP status 304 - No news from last check", nil)];
    [connectorItem setStatus:NSLocalizedString(@"No new articles available", nil)];
});
```

This seems to stop the errors and show the result much quicker too. Could this be the problem?

**Comment by barijaona** [comment]
Can anyone have a look at my branch [issue833ter](https://github.com/barijaona/vienna-rss/tree/issue833ter) ?
It undoes parts of pull request #864 that @josh64x2 does not like and uses a different approach.

What I don't understand at all is : 
- why the following scheme does not fix the issue : 

````objective-c
dispatch_async(_queue, ^() {
    ...
    [anObject theMethod];
    ...
});

-(void)theMethod {
	dispatch_async(dispatch_get_main_queue(), ^{
	    // update value which updates UI + other stuff
        ...        
	});
}
````

- while the following scheme seems OK ?

```objective-c
dispatch_async(_queue, ^() {
    ...
    dispatch_async(dispatch_get_main_queue(), ^{
        [anObject theMethod];
    });    
    ...
});

-(void)theMethod {
    // update value which updates UI + other stuff
    ...        
}
````

They seem functionally equivalent to me, but obviously aren't. I lost hours with the first scheme because it was less verbose, while the second scheme was immediately OK…

**Comment by barijaona** [comment]
Fixed by #890
